src/main/java/duke/command/AddDeadlineCommand.java

```diff
@@ -1,23 +1,30 @@
 package duke.command;
 
+import duke.command.exceptions.InvalidParameterError;
+import duke.command.utils.DateTimeStringParser;
 import duke.interfaces.Command;
 import duke.interfaces.View;
 import duke.model.Task;
 import duke.model.TaskModel;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
 
 public class AddDeadlineCommand implements Command {
     private final TaskModel taskModel;
     private final View taskView;
     private final String todoDescription;
-    private final String deadline;
+    private final LocalDateTime deadline;
     private static final String addedTaskMessage = "Got it. I've added this task:\n  ";
-    AddDeadlineCommand(View taskView, TaskModel taskModel, String todoDescription, String deadline) {
+    AddDeadlineCommand(View taskView, TaskModel taskModel, String todoDescription, String deadline)
+            throws InvalidParameterError {
         this.todoDescription = todoDescription;
-        this.deadline = deadline;
         this.taskView = taskView;
         this.taskModel = taskModel;
+        LocalTime defaultDeadlineTime = LocalTime.MAX;
+        this.deadline = DateTimeStringParser.parseDateTimeString(deadline, defaultDeadlineTime);
```

Comment:
Hi, I'm your reviewer. I like the implementation of a failsafe for the deadline. Very elegant way to handle missing information.

Reply:
Thanks, appreciate the input!